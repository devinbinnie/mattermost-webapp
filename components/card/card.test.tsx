// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

import Card from './card';
import CardBody from './card_body';

describe('components/card/card', () => {
    const baseProps = {
        expanded: false,
    };

    test('should match snapshot', () => {
        const wrapper = mount(
            <Card {...baseProps}>
                <Card.Header {...baseProps}>{'Header Test'}</Card.Header>
                <Card.Body {...baseProps}>{'Body Test'}</Card.Body>
            </Card>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should have a height based on content when expanded', () => {
        const wrapper: ReactWrapper<any, any, CardBody> = mount(
            <Card.Body {...baseProps}>
                {'Body Test'}
                {'Slightly Larger Body Text'}
            </Card.Body>
        );

        expect(wrapper.instance().card.current?.style.height).toBe('');

        wrapper.setProps({expanded: true});
        expect(wrapper.instance().card.current?.style.height).not.toBe('');

        wrapper.setProps({expanded: false});
        expect(wrapper.instance().card.current?.style.height).toBe('');
    });
});
