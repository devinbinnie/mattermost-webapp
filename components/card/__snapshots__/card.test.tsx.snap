// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/card/card should match snapshot 1`] = `
<Card
  expanded={false}
>
  <div
    className="Card"
  >
    <CardHeader
      expanded={false}
      key=".0"
    >
      <div
        className="Card__header"
      >
        Header Test
      </div>
    </CardHeader>
    <CardBody
      expanded={false}
      key=".1"
    >
      <div
        className="Card__body"
      >
        Body Test
      </div>
    </CardBody>
  </div>
</Card>
`;
